tests/e2e/test_couchbase.py

```diff
@@ -0,0 +1,328 @@
+        )
+
+    def is_up_and_running(self, pod_name, ocp_value):
+        """
+        Test if the pod specified by pod is up and running.
+        """
+        if not pod_name:
+            return False
+        pod_info = ocp_value.exec_oc_cmd(f"get pods {pod_name} -o json")
+        if pod_info['status']['containerStatuses'][0]['ready']:
+            if 'running' in pod_info['status']['containerStatuses'][0]['state']:
+                return True
+        return False
+
+    def parse_couchbase_log(self, pf_pod):
+        """
+        Run oc logs on the pillowfight pod passed in.  Cleanup the output
+        from oc logs to handle peculiarities in the couchbase log results,
+        and generate a summary of the results.
+
+        Returns: a dictionary with two values; 'opspersec' and 'resptimes'.
+        opspersec is a list of ops per second numbers reported.'
+        resptimes is a dictionary index by the max response time of a range.
+        Each entry in resptimes contains a minimum response time for that range,
+        and a count of how many messages fall within that range.
+        """
+        data_from_log = self.up_check.exec_oc_cmd(
+            f"logs -f {pf_pod} --ignore-errors",
+            out_yaml_format=False
+        )
+        # The data in the couchbase logs is kind of abnormal.
+        # It contains histograms with invalid unicode charaters for yaml
+        # output (which is why out_yaml_format=False is used).
+        # It also seems to write a block of text inside another block at
+        # an unpredictable location.  The value good_txt below is the output
+        # of the log with that data removed..
+        #
+        # So what's left is a list of OPS/SEC values and a histogram of
+        # response times.  This routine organizes that data.
+        ops_per_sec = []
+        resp_hist = {}
+        end_okay_txt = data_from_log.rfind('\n{')
+        start_okay_txt = data_from_log.rfind('orph: 0\n') + 8
+        raw_txt = data_from_log[:end_okay_txt] + data_from_log[start_okay_txt:]
+        good_txt = raw_txt.replace("\0", "")
+        lines = good_txt.split("\n")
+        collect_response_times = False
+        for dline in lines:
+            if dline.startswith("OPS/SEC"):
+                dfields = dline.split(" ")
+                dnumb = int(dfields[-1].strip())
+                ops_per_sec.append(dnumb)
+            if collect_response_times:
+                if dline.startswith('['):
+                    for element in ["[", "]", "|", "-", "#"]:
+                        dline = dline.replace(element, " ")
+                    parts = dline.split()
+                    i1 = int(parts[0])
+                    i2 = int(parts[1])
+                    if parts[2] == 'ms':
+                        i1 *= 1000
+                        i2 *= 1000
+                    resp_hist[i2] = {'minindx': i1, 'number': int(parts[3])}
+            if "+---------+" in dline:
```

Comment:
Does this mean if they change logging this will stop working ?

Reply:
If the format changes enough, this will not work.   I'll try to use regular expressions for this checking.

Reply:
This will be handled in the next checkin